src/formatDate.js

```diff
@@ -9,6 +43,38 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+
+  const possibleDateParts = ['YYYY', 'YY', 'MM', 'DD'];
+
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  const oldDateParts = date.split(oldSeparator);
+
+  let datePartsMatched = {};
```

Comment:
The variable `datePartsMatched` is declared as an empty object but it's not immediately clear that it will be used to store date parts. A comment explaining its purpose or a more descriptive name could improve code readability.